### PR TITLE
Handles 856-field.

### PR DESCRIPTION
**Pull request by natlibesa**

### CODE REVIEW

File: src/transform/convert/index.js [hunk 1]
```diff
@@ -34,6 +34,11 @@ import generateStaticFields from './generate-static-fields';
...
 import {createLogger} from '@natlibfi/melinda-backend-commons';
 
+
+import fetch from 'node-fetch';
+const URN_GENERATOR_URL = 'http://generator.urn.fi/cgi-bin/urn_generator.cgi?type=nbn';
+
+
 const logger = createLogger();
 
 export default ({source4Value, isLegalDeposit, sources, sender, moment = momentOrig}) => ({Product: record}) => {
```
Comment (natlibfi-arlehiko) [inline src/transform/convert/index.js:44]: Change to `export default ({source4Value, isLegalDeposit, sources, sender, moment = momentOrig}) => async ({Product: record}) => {`

File: src/transform/convert/index.js [hunk 2]
```diff
@@ -153,7 +158,7 @@ export default ({source4Value, isLegalDeposit, sources, sender, moment = momentO
       generate653(),
       generate655(),
       generate700(),
-      // Generate856(),  // waits
+      generate856(), // <--
```
Comment (natlibfi-arlehiko) [inline src/transform/convert/index.js:161]: Change to `await generate856`

File: src/transform/convert/index.js [hunk 3]
```diff
@@ -764,30 +769,48 @@ export default ({source4Value, isLegalDeposit, sources, sender, moment = momentO
...
+        const isbn13 = getValues('ProductIdentifier').find(({ProductIDType: [type]}) => type === '15');
+
+        if (isbn13) {
+          return isbn13.IDValue[0];
+        }
+
+        return getValues('ProductIdentifier').find(({ProductIDType: [type]}) => type === '02')?.IDValue?.[0];
+      }
+
+      function createURN(isbn = false) {
```
Comment (natlibfi-arlehiko) [inline src/transform/convert/index.js:802]: Change to `await function createURN(isbn = false) {`

File: src/transform/convert/index.js [hunk 4]
```diff
@@ -764,30 +769,48 @@ export default ({source4Value, isLegalDeposit, sources, sender, moment = momentO
...
+        const isbn = getIsbn();
 
         return [
           {
             tag: '856',
             ind1: '4',
             ind2: '0',
             subfields: [
-              {code: 'u', value: 'URN of legal deposit XXXXXXXXXX under construction'},
+              {code: 'u', value: createURN(isbn)},
```
Comment (natlibfi-arlehiko) [inline src/transform/convert/index.js:784]: Change to `{code: 'u', value: await createURN(isbn)},`

File: src/transform/convert/index.js [hunk 5]
```diff
@@ -764,30 +769,48 @@ export default ({source4Value, isLegalDeposit, sources, sender, moment = momentO
...
+
+        return getValues('ProductIdentifier').find(({ProductIDType: [type]}) => type === '02')?.IDValue?.[0];
+      }
+
+      function createURN(isbn = false) {
+        if (isbn) {
+          return `http://urn.fi/URN:ISBN:${isbn}`;
+        }
+
+        const response = fetch(URN_GENERATOR_URL);
```
Comment (natlibfi-arlehiko) [inline src/transform/convert/index.js:807]: Change to `const response = await fetch(URN_GENERATOR_URL);`

File: src/transform/convert/index.js [hunk 6]
```diff
@@ -764,30 +769,48 @@ export default ({source4Value, isLegalDeposit, sources, sender, moment = momentO
...
+        const body = response.text();
```
Comment (natlibfi-arlehiko) [inline src/transform/convert/index.js:808]: Change to `const body = await response.text();`

**Review by natlibfi-arlehiko — CHANGES_REQUESTED** [review]

File: src/transform/convert/index.js [hunk 7]
```diff
@@ -818,13 +818,13 @@ export default ({source4Value, isLegalDeposit, sources, sender, moment = momentO
         return getValues('ProductIdentifier').find(({ProductIDType: [type]}) => type === '02')?.IDValue?.[0];
       }
 
-      function createURN(isbn = false) {
+      await function createURN(isbn = false) {
```
Comment (natlibfi-jonollil) [inline src/transform/convert/index.js:821]: async function...

File: src/transform/convert/index.js [hunk 8]
```diff
@@ -800,15 +800,15 @@ export default ({source4Value, isLegalDeposit, sources, sender, moment = momentO
...
+              {code: 'u', value: await createURN(isbn)},
               {code: 'z', value: 'Käytettävissä vapaakappalekirjastoissa'},
               {code: '5', value: 'FI-Vapaa'}
             ]
           }
         ];
       }
 
-      function getIsbn() {
+      function getIsbn() {
```
Comment (natlibfi-jonollil) [inline src/transform/convert/index.js:811]: In VScode: File -> Preferences -> Settings -> search: trim -> enable: Trailing-spaces: Trim On Save

File: src/transform/convert/generate3XXFields.js [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+  function getMinutes() {
+
+    if (extValue.slice(3, 5).replace(/0/gu, '') === '') { // eslint-disable-line functional/no-conditional-statement
+      return '';
+    }
+    return `${extValue.slice(3, 5).replace(/0/gu, '')} min`;
+  }
+
+
```
Comment (natlibfi-jonollil) [inline src/transform/convert/generate3XXFields.js:88]: Empty lines clean up. 1 empty between functions and 0 between } }. Groupping up the consts and one emptyline before if makes code more readable

**Review by natlibfi-jonollil — CHANGES_REQUESTED** [review]

**Review by natlibfi-jonollil — APPROVED** [review]

**Review by natlibfi-jonollil — APPROVED** [review]

**Pull request merged**